adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/uap/PlatformInformation.cs

```diff
@@ -53,21 +56,69 @@ public override string GetEnvironmentVariable(string variable)
             return localSettings.Values.ContainsKey(variable) ? localSettings.Values[variable].ToString() : null;
         }
 
+        /// <summary>
+        /// Get a principal name that can be used for WIA
+        /// </summary>
+        /// <remarks>
+        /// Win10 allows several identities to be logged in at once; 
+        /// select the first principal name that can be used
+        /// </remarks>
+        /// <returns></returns>
         public override async Task<string> GetUserPrincipalNameAsync()
         {
-            if (!UserInformation.NameAccessAllowed)
+            IReadOnlyList<User> users = await User.FindAllAsync();
+            if (users == null || !users.Any())
             {
-                throw new AdalException(AdalErrorEx.CannotAccessUserInformation, AdalErrorMessageEx.CannotAccessUserInformation);
+                throw new AdalException(
+                 AdalErrorEx.CannotAccessUserInformationOrUserNotDomainJoined,
+                 CoreErrorMessages.UapCannotFindDomainUser);
             }
 
-            try
+            var getUserDetailTasks = users.Select(async u =>
             {
-                return await UserInformation.GetPrincipalNameAsync().AsTask().ConfigureAwait(false);
+                object domainObj = await u.GetPropertyAsync(KnownUserProperties.DomainName);
+                string domainString = domainObj?.ToString();
+
+                object principalObject = await u.GetPropertyAsync(KnownUserProperties.PrincipalName);
+                string principalNameString = principalObject?.ToString();
+
+                return new { Domain = domainString, PrincipalName = principalNameString };
+            }).ToList();
+
+            var userDetails = await Task.WhenAll(getUserDetailTasks).ConfigureAwait(false);
+
+            // try to get a user that has both domain name and upn
+            var userDetailWithDomainAndPn = userDetails.FirstOrDefault(
+                d => !String.IsNullOrWhiteSpace(d.Domain) &&
+                !String.IsNullOrWhiteSpace(d.PrincipalName));
+
+            if (userDetailWithDomainAndPn != null)
+            {
+                return userDetailWithDomainAndPn.PrincipalName;
```

Comment:
Exceptions are not thrown if the app manifest does not have the right capabilities, the properties returned are empty